packages/web/src/scss/components/FileUploader/_FileUploaderInput.scss

```diff
@@ -85,10 +86,54 @@ $_field-name: 'FileUploaderInput';
     display: initial;
 }
 
+:is(.FileUploaderInput--disabled, .FileUploaderInput.is-disabled)
+    :is(.FileUploaderInput__message, [data-element='validator_message']) {
+    @include form-fields-tools.message-disabled();
+}
+
+:is(.FileUploaderInput--disabled, .FileUploaderInput.is-disabled) :is(.FileUploaderInput__helperText) {
+    @include form-fields-tools.helper-text-disabled();
+}
+
+// 3.
+// stylelint-disable selector-max-compound-selectors, selector-max-specificity
+:is(.FileUploaderInput--disabled, .FileUploaderInput.is-disabled)
+    > .FileUploaderInput__input
+    ~ .FileUploaderInput__dropZone
+    > svg,
+:is(.FileUploaderInput--disabled, .FileUploaderInput.is-disabled)
+    > .FileUploaderInput__input
+    ~ .FileUploaderInput__dropZone
+    > .FileUploaderInput__dropZoneLabel,
+:is(.FileUploaderInput--disabled, .FileUploaderInput.is-disabled)
+    > .FileUploaderInput__input
+    ~ .FileUploaderInput__dropZone
+    > .FileUploaderInput__dropZoneLabel
+    > .FileUploaderInput__link {
+    @include form-fields-tools.input-disabled();
```

Comment:
The mixin adds styles for `<input />`, but here you have `dropZoneLabel`, `link`, and even SVG.

- Do they really use the same tokens?
- Will it last? That is, are we sure (enough) it doesn't break in the future when someone changes the `input-disabled()` mixin, without knowing it's used by something else than inputs (= the drop zone)?

I am in doubts and would consider detaching these elements from the input mixin. What do you think? 🤔

Reply:
I refactored it. Simplified the code and focused on based styling mainly on the disabled input. Where it made sense I avoided mixin. See the latest fixup.